### PR TITLE
WL-4148 Added email domain validator in profile tool.

### PR DESCRIPTION
**Pull request by ouit0408**
Created new EmailValidator class for validating email in profile.It uses
Wicket's EmailAddressValidator to check the format of email and Kernel's
Validator class to check for the valid domain.

Removed duplicate 'official.user.name' from admin.properties in user tool.

### CODE REVIEW

File: profile2/tool/src/java/org/sakaiproject/profile2/tool/components/EmailValidator.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		Map map = super.variablesMap(validatable);
+		//Check first if the email address is of correct format
+		if (isValidEmailAddress(emailText)) {
+			//check if email address contains any of accepted local/official hosts
+			if(!Validator.isAllowedLocalEmailDomain(emailText)){
+				map.put("invalidAccount", ServerConfigurationService.getString("invalidNonOfficialAccountString"));
+				map.put("localAccount", ServerConfigurationService.getString("localUserAccount", OFFICIAL_USER_NAME));
+				error(validatable, getClass().getSimpleName() + ".profile.emailbaddomain", map);
+			}
+		}
```
Comment (buckett) [inline profile2/tool/src/java/org/sakaiproject/profile2/tool/components/EmailValidator.java]: Could you match the style of the project.

Comment (ouit0408) [reply profile2/tool/src/java/org/sakaiproject/profile2/tool/components/EmailValidator.java]: made the change.

**Comment by buckett** [comment]
Is this new code for 11.x or a missing commit from the migration?

**Comment by ouit0408** [comment]
this is a new code as the validation was never done for profile tool .it was done for accounts, user and site-info tool in 10

File: profile2/tool/src/java/org/sakaiproject/profile2/tool/components/EmailValidator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	private static final long serialVersionUID = 1L;
+	private static final String OFFICIAL_USER_NAME = "Official Username";
+
+	public void onValidate(IValidatable validatable) {
+		String emailText = (String)validatable.getValue();
+		Map map = super.variablesMap(validatable);
+		//Check first if the email address is of correct format
+		if (isValidEmailAddress(emailText)) {
+			//check if email address contains any of accepted local/official hosts
+			if(!Validator.isAllowedLocalEmailDomain(emailText)) {
```
Comment (buckett) [inline profile2/tool/src/java/org/sakaiproject/profile2/tool/components/EmailValidator.java:27]: I think for going back to Sakai the code in `Validator` will have to be moved as it's deprecated, but maybe we shouldn't do this now.

**Pull request merged**